library/core-components/components/nav-item/nav-item.styles.ts

```diff
@@ -8,18 +8,18 @@ export type StyleProps = Pick<
 >;
 
 export const useStyles = ({ selected }: StyleProps) => ({
-  link: css`
-    // TODO: remove this once we have new destructured typography object containing this property
+  styles: css`
+    // TODO: remove this once we have new destructured typography object containing this property, or add this to a global reset
     text-decoration: none;
 
     &:hover {
       // apply label color when link is hovered
-      > span {
+      .nav-item-label {
         color: var(${radiusTokens.component.color.navigationItem.hover.label});
       }
 
       // apply underline color when link is hovered but not selected
-      > div {
+      .nav-item-underline {
```

Comment:
I tried to avoid adding hard-coded classnames but it was necessary to cleanly target the desired elements in this case. Apparently previous versions of `emotion` provided the ability to use generated classnames in this way, but that has been removed in newer versions for some reason... see discussion here (little movement on this by the looks of it): https://github.com/emotion-js/emotion/issues/1217